components/brave_wallet_ui/components/extension/connected-panel/index.tsx

```diff
@@ -122,6 +123,13 @@ const ConnectedPanel = (props: Props) => {
 
   const onClickViewOnBlockExplorer = useExplorer(selectedNetwork)
 
+  const showConnectButton = React.useMemo((): boolean => {
+    if (originInfo.origin) {
+      return originInfo.origin.startsWith('chrome://')
+    }
+    return false
+  }, [originInfo])
```

Comment:
Passing objects into a memo usually causes the memo to recompute if it is not the exact object reference. 
Refactoring to check only for originInfo.origin would check against a primitive value and recompute less. 
Consider that, or replacing the memo with `const showConnectButton = originInfo?.origin?.startsWith('chrome://')`

Reply:
Removed this memo and doing check directly like you recommended in your first comment!